ingestors/kafka/src/main/scala/hydra/kafka/endpoints/TopicMetadataEndpoint.scala

```diff
@@ -171,9 +171,15 @@ class TopicMetadataEndpoint[F[_]: Futurable](consumerProxy:ActorSelection,
                         Futurable[F].unsafeToFuture(createTopicProgram
                           .createTopicFromMetadataOnly(t, req))
                       ) {
-                        case Failure(exception) =>
-                          addHttpMetric(topic, StatusCodes.InternalServerError, "/v2/metadata", startTime, method.value)
-                          complete(StatusCodes.BadRequest, s"Unable to create Metadata for topic $topic : ${exception.getMessage}")
+                        case Failure(exception) => exception match {
+                          case e:IncompatibleSchemaException =>
+                            addHttpMetric(topic, StatusCodes.BadRequest, "/v2/metadata", startTime, method.value, error=Some(e.getMessage))
+                            complete(StatusCodes.BadRequest, s"Unable to create Metadata for topic $topic : ${exception.getMessage}")
+                          case _ =>
+                            addHttpMetric(topic, StatusCodes.InternalServerError, "/v2/metadata", startTime, method.value)
```

Comment:
You need to add the error=Some(e.getMessage) here as well. So you can either name the case class, or like you have on line 180, use the exception you are matching on.

Reply:
Done